src/main/java/software/amazon/awssdk/crt/mqtt/MqttConnectionEvents.java

```diff
@@ -0,0 +1,38 @@
+
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package software.amazon.awssdk.crt.mqtt;
+
+/** 
+ * Interface used to receive connection events from the CRT 
+ */
+public interface MqttConnectionEvents {
+    /**
+     * called when a connection is resolved, one way or the other. Non-zero error
+     * code indicates connection never completed or was disconnected
+     */
+    void onConnectionComplete(int errorCode, boolean sessionPresent);
```

Comment:
In python, we're just exposing connect-completion via the Future returned from connect().

Is there a compelling reason to offer the user 2 ways to learn of connect-completion (via this callback and also via the CompletableFuture)?

Also, if there's a compelling reason for connect-completion, why isn't there a disconnect-completion?

Reply:
There isn't, I guess I could re-work this to pull that out.